Tests/SwiftLintFrameworkTests/RulesTests.swift

```diff
@@ -62,6 +62,26 @@ class RulesTests: XCTestCase {
         verifyRule(DiscardedNotificationCenterObserverRule.description)
     }
 
+    func testDiscouragedInit() {
```

Comment:
I think it makes sense to move everything to the new file. See `FileHeaderRuleTests.swift` for example.